storage/innobase/handler/ha_innodb.cc

```diff
@@ -6755,13 +6756,17 @@ wsrep_store_key_val_for_row(
 			/* cannot exceed max column lenght either, we may need to truncate
 			the stored value: */
 			if (true_len > sizeof(sorted)) {
-			  true_len = sizeof(sorted);
+				true_len = sizeof(sorted);
 			}
 
-			memcpy(sorted, data, true_len);
-			true_len = wsrep_innobase_mysql_sort(
-				mysql_type, cs->number, sorted, true_len,
-				REC_VERSION_56_MAX_INDEX_COL_LEN);
+			ut_ad(data || (data == nullptr && true_len == 0));
```

Comment:
I find it very confusing to read logic expressions that contain such redundancy. `data == nullptr` can be written shorter as just `!data`. In that form, it is clearer to understand the redundancy: `data || (!data && true_len == 0)`. Only one of `data` and `!data` can hold at any given time. Due to short-circuit evaluation, the right-hand-side of `data || true_len == 0` is evaluated only if `!data`. Therefore, an additional condition about `data`, as in `(!data && true_len == 0)`, would be redundant. It would even be the case if there was no short-circuit evaluation taking place.

Reply:
Fixed.